### PR TITLE
Update examples hello

### PR DESCRIPTION
**Pull request by subfuzion**
Updates hello example to latest packages (depends on updated functions_framework and functions_framework_builder packages).

Please merge #193 and #194 first.

### CODE REVIEW

File: examples/hello/pubspec.yaml [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
-  functions_framework_builder: ^0.3.0
-  http: ^0.12.0
-  test: ^1.15.0
-  test_process: ^1.0.0
+  functions_framework_builder: ^0.4.0
+  http: ^0.13.1
+  test: ^1.16.8
+  test_process: ^2.0.0
+
+dependency_overrides:
```
Comment (kevmoo) [inline examples/hello/pubspec.yaml:21]: need to remove all of this

File: functions_framework/pubspec.yaml [hunk 2]
```diff
@@ -10,18 +10,18 @@ environment:
 dependencies:
```
Comment (kevmoo) [inline functions_framework/pubspec.yaml:10]: revert this change.

File: functions_framework_builder/pubspec.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ environment:
   sdk: ">=2.12.0 <3.0.0"
 
 dependencies:
-  analyzer: ^1.0.0
+  analyzer: ^1.2.0
```
Comment (kevmoo) [inline functions_framework_builder/pubspec.yaml:10]: revert this change

File: examples/hello/Makefile [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# For illustrative and convenience purposes only; you don't have to use make,
```
Comment (kevmoo) [inline examples/hello/Makefile:1]: This seems a bit...weird...

Comment (subfuzion) [reply examples/hello/Makefile:1]: The comment, or the Makefile? I use the Makefile myself, which is why I decided to contribute it. But we can remove. I can also document how to create simple little aliases or bash functions to do the same thing as a memory aid.

File: examples/hello/pubspec.yaml (hunk 1 above)
Comment (grant) [reply examples/hello/pubspec.yaml:21]: @kevmoo Can you explain why?

File: functions_framework/pubspec.yaml (hunk 2 above)
Comment (grant) [reply functions_framework/pubspec.yaml:10]: @kevmoo Can you explain why?

File: functions_framework_builder/pubspec.yaml (hunk 3 above)
Comment (grant) [reply functions_framework_builder/pubspec.yaml:10]: @kevmoo Can you explain why?

Comment (kevmoo) [reply functions_framework_builder/pubspec.yaml:10]: It's a bit of a wash. Since we're only testing w/ latest we should likely pin latest.

...there is a problem where someone using our code has explicitly excluded `analyzer v1.2` because they are avoiding a change. In which case we'll have dependency lock.

I generally only bump this if I know I'm using a new feature. 🤷

File: functions_framework/pubspec.yaml (hunk 2 above)
Comment (kevmoo) [reply functions_framework/pubspec.yaml:10]: The scope of this PR should just be examples, right? Not changing the published packages – at least that's what I thought
